botorch/optim/core.py

```diff
@@ -202,7 +202,7 @@ def torch_minimize(
         )
 
         # TODO: Update stopping_criterion API to return a message.
-        if stopping_criterion and stopping_criterion(fval):
+        if stopping_criterion and stopping_criterion(fval.detach()):
```

Comment:
It looks like `fval` is detached everywhere it's used. Perhaps line 195 should be `fval = closure()[0].detach()`?

Reply:
yeah that would be better

Reply:
Good idea.